hal_st/synchronous_stm32fxxx/SynchronousAesStm.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef HAL_SYNCHRONOUS_AES_STM_HPP
+#define HAL_SYNCHRONOUS_AES_STM_HPP
+
+#include DEVICE_HEADER
+#include "generated/stm32fxxx/PeripheralTable.hpp"
+#include "infra/util/ByteRange.hpp"
+
+#if defined(HAS_PERIPHERAL_AES)
+
+namespace hal
+{
+    class SynchronousAesEcbStm
+    {
+    public:
+        enum class DataSwapping : uint8_t
```

Comment:
What is data swapping actually? Is it relevant for your proposition? Otherwise, you could just remove it.
If you keep DataSwapping, please consider spending a comment on clarifying what it does. Or maybe there's a better way :)